log/field.go

```diff
@@ -0,0 +1,201 @@
+package log
+
+import "math"
+
+type fieldType int
+
+const (
+	stringType fieldType = iota
+	boolType
+	intType
+	int32Type
+	uint32Type
+	int64Type
+	uint64Type
+	float32Type
+	float64Type
+	errorType
+	objectType
+	lazyLoggerType
```

Comment:
I was wondering if we could remove key from the Encoder interface, but I
fear that doing so ultimately requires using an interface type as the Field
type, which will hurt performance.

When you ask about nested fields, I start to wonder exactly how you'll use
it.

Another question: I don't see slices or array types in the list of
supported types. Omission or intentional?

ᐧ

On Tue, Aug 30, 2016 at 6:38 PM, Yuri Shkuro <notifications@github.com>
wrote:

> In log/field.go
> <https://github.com/opentracing/opentracing-go/pull/108#discussion_r76911015>
> :
>
> > +
> > +type fieldType int
> > +
> > +const (
> > +	stringType fieldType = iota
> > +	boolType
> > +	intType
> > +	int32Type
> > +	uint32Type
> > +	int64Type
> > +	uint64Type
> > +	float32Type
> > +	float64Type
> > +	errorType
> > +	objectType
> > +	lazyLoggerType
>
> one other interesting feature in zap was nested fields, which allow
> logging hierarchical structured data.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/opentracing/opentracing-go/pull/108/files/64b19712cad0dd3fd4a3705aef7e7e8b727ba697#r76911015>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADdiiSwNEJvns4ccZ1B52n-rjALQlR5Cks5qlNr7gaJpZM4JvDpj>
> .
>